### PR TITLE
Database

### PR DESCRIPTION
**Pull request by RachelPun**
fixed connect_db.sh

### CODE REVIEW

File: judges/pipeline.py [hunk 1]
```diff
@@ -1,13 +1,199 @@
...
+        for c in range(0, len(cells), 2):
+            rows.append({"judge": cells[c],
+                         "appointment": cells[c+1]})
+
+        return pd.DataFrame(rows)
+
+    except requests.RequestException as error:
+        logging.info(f"Error fetching URL: {error}")
+        return pd.DataFrame([])
+
```
Comment (MahinRahman8901) [inline judges/pipeline.py:39]: Good use of try and except to catch errors

File: judges/pipeline.py [hunk 2]
```diff
@@ -1,13 +1,199 @@
...
+    datestr = datestr.split()[0]
+    length = len(datestr)
+
+    if length == 10:
+        return datetime.strptime(datestr, "%d-%m-%Y")
+    if length == 8:
+        return datetime.strptime(datestr, "%d-%m-%y")
+    if length == 9:
+        return datetime.strptime(datestr, "%d-%b-%y")
```
Comment (MahinRahman8901) [inline judges/pipeline.py:81]: Why are there varying numbers when comparing the length of the datetime?

File: judges/pipeline.py [hunk 3]
```diff
@@ -1,13 +1,199 @@
...
+def transform_df(df: pd.DataFrame,
+                 title: str,
+                 type: str,
+                 circuit: str = "") -> pd.DataFrame:
+    """Cleans and enriches the data.
+    Returns transformed data as pd.DF."""
+
+    df["appointment"] = df["appointment"].apply(convert_date)
+
+    df["judge"] = df["judge"].apply(extract_name, args=(title,))
```
Comment (MahinRahman8901) [inline judges/pipeline.py:120]: great use of .apply to run the function on the entire column in the df

File: judges/pipeline.py [hunk 4]
```diff
@@ -1,13 +1,199 @@
...
+
+
+# ========== MAIN ==========
 if __name__ == "__main__":
 
+    HCKB_URL = "https://www.judiciary.uk/about-the-judiciary/who-are-the-judiciary/senior-judiciary-list/kings-bench-division-judges/"
+    CJ_URL = "https://www.judiciary.uk/about-the-judiciary/who-are-the-judiciary/list-of-members-of-the-judiciary/circuit-judge-list/"
+
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(encoding='utf-8', level=logging.INFO)
```
Comment (MahinRahman8901) [inline judges/pipeline.py:175]: Love the fact you logged everything as well, good job!

**Review by MahinRahman8901 — APPROVED** [review]

File: judges/pipeline.py (hunk 2 above)
Comment (RachelPun) [reply judges/pipeline.py:81]: because there's sadly a mix of formats on the government website

**Pull request merged**